### ISSUE TITLE
Narrow no-break space (U+202F) incorrectly rendered.

### ISSUE THREAD

**Issue by cjns1989** [koreader/koreader#6987]
#### Context:

* KOReader version: v2019.04-27-g7e06db9_2019-04-14
* Device: Onyx Boox Poke 3

#### How to recreate:

Load an EPUB with thin no-break spaces (correctly typeset French EPUB's should have many of those)

#### Description:

There are a variety of rules that have to be followed when typesetting French. One of them is that punctuation that is made up of 2 separate parts such as the semi-colon (;), the colon (:), the question mark (?), the French quotes kown as 'guillemots' (« ») … etc. should not follow immediately after the preceding word: a narrow space should keep the two visually separate. In the digital world this is done by inserting a narrow no-break space (U+202F) between the two.

I am looking at an EPUB that happens to be in French and all the colons, semi-colons, question marks… etc. show up like so:

Bonjour?! comment-allez-vous??
Très bien?; et vous??

In other words all the U+202F's in the source files are rendered by '?''s (question marks).

I looked for some way work around the issue   i.e.  checking the encoding in the settings (it should be UTF-8 in this case) but could find anything in this respect … switched to different fonts and couldn't find a way to  correct this.

I have tried the same EPUB in a variety of Android eReader apps (Readera, Aldiko, Gitden, PocketBook, FBReader…) and Koreader appears to be the only eReading application having the problem.

Is this something caused by my setup or is it a known issue?

**Comment by NiLuJe** [comment]
The ridiculously old version you're using probably doesn't help ;).

If you can still reproduce this in either the current stable or nightly, we're going to need an actual snippet of code.

One simple way to do that would be to highlight a section of affected text, and tap "View HTML"

(@poire-z may then ask you for a more specific debug view screenshot, or not ^^).

**Comment by poire-z** [comment]
> Is this something caused by my setup or is it a known issue?
> The ridiculously old version you're using probably doesn't help ;).

Probably :) 
Just tested with U+202F or `&#x202F;` in an HTML and in an EPUB file: no issue for me, with any of the `Font kerning` option, and with various fonts.
![image](https://user-images.githubusercontent.com/24273478/102007167-fac1f980-3d26-11eb-8e5e-1d7520f64240.png)

But I do get `?` with various main fonts if I disable `Font settings > [ ] Use additional fallback fonts`:
![image](https://user-images.githubusercontent.com/24273478/102007164-ebdb4700-3d26-11eb-8b4d-d998b417a546.png)
